### PR TITLE
Set/Get TextureData and corrections

### PR DESCRIPTION
**Pull request by cybik**

### CODE REVIEW

File: src/FNA3D_Driver_OpenGL.c [hunk 1]
```diff
@@ -2238,6 +2238,77 @@ static void OPENGL_INTERNAL_DisposeBackbuffer(OpenGLDevice *device)
 	#undef GLBACKBUFFER
 }
 
+uint8_t OPENGL_INTERNAL_ReadTargetIfApplicable(
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_OpenGL.c]: This can be static

File: src/FNA3D_Driver_OpenGL.c [hunk 2]
```diff
@@ -2497,6 +2577,15 @@ int OPENGL_INTERNAL_Texture_GetFormatSize(FNA3D_SurfaceFormat format)
 	}
 }
 
+int OPENGL_INTERNAL_Texture_GetPixelStoreAlignment(FNA3D_SurfaceFormat format) {
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_OpenGL.c]: Static... this applies to all internal functions

File: src/FNA3D_Driver_OpenGL.c [hunk 3]
```diff
@@ -2553,7 +2642,7 @@ FNA3D_Texture* OPENGL_CreateTexture2D(
 		}
 	}
 
-	return result;
+	return (FNA3D_Texture*)result;
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_OpenGL.c]: Space between cast and value

File: src/FNA3D_Driver_OpenGL.c [hunk 4]
```diff
@@ -2695,58 +2840,211 @@ void OPENGL_SetTextureData3D(
...
 
 void OPENGL_SetTextureDataCube(
 	void* driverData,
 	FNA3D_Texture *texture,
 	FNA3D_SurfaceFormat format,
-	int32_t x,
-	int32_t y,
-	int32_t w,
-	int32_t h,
+	int32_t xOffset,
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_OpenGL.c]: Keep these as they were if possible, probably more work...

File: src/FNA3D_Driver_OpenGL.c (hunk 1 above)
Comment (cybik) [reply src/FNA3D_Driver_OpenGL.c]: argh

**Comment by flibitijibibo** [comment]
Definitely going to need a VS2010 pass in the morning...

**Pull request merged**